### PR TITLE
Scoring logic

### PR DESCRIPTION
**Pull request by sunnyalgo**
Scoring Logic Implemented.

### CODE REVIEW

File: contracts/Recruitment.sol [hunk 1]
```diff
@@ -38,30 +38,26 @@ contract Recruitment is Ownable, ReentrancyGuard {
...
+    mapping(address => FrontDoorStructs.UserScore[]) public referralScores;
+    mapping(address => FrontDoorStructs.UserScore[]) public companyScores;
+    mapping(address => FrontDoorStructs.UserScore[]) public candidateScores;
+
     mapping(address => mapping(address => bool)) public hasScoredCompany; //allows only to score once
     mapping(address => bool) public isCompany; // check if company is registered or not
 
     mapping(uint256 => FrontDoorStructs.Candidate[]) public candidateListForJob; // list of candidates for a job
-    mapping(uint256 => FrontDoorStructs.Candidate) public jobCandidatehire;
+    mapping(uint256 => FrontDoorStructs.Candidate[]) public hiredCpandidateListForJob;
```
Comment (codingnuclei) [inline contracts/Recruitment.sol]:
```suggestion
    mapping(uint256 => FrontDoorStructs.Candidate[]) public hiredCandidateForJobList;
```

File: contracts/Recruitment.sol [hunk 2]
```diff
@@ -115,7 +111,19 @@ contract Recruitment is Ownable, ReentrancyGuard {
     function isCompanyRegistered(address _company) public view returns (bool) {
         return isCompany[_company];
     }
-
+    function registerCandidate() external {
+      FrontDoorStructs.Candidate memory newCandidate = FrontDoorStructs.Candidate(
```
Comment (codingnuclei) [inline contracts/Recruitment.sol:121]: check not already registered.

File: contracts/Recruitment.sol [hunk 3]
```diff
@@ -278,7 +286,7 @@ contract Recruitment is Ownable, ReentrancyGuard {
             referralList[_referralCounter].referralCode == _referralCode,
             "Invalid Referral Code"
         ); // check if referral code is valid for referral
-
+        confirmedReferralCount[candidateList[msg.sender].referrer]++;
```
Comment (codingnuclei) [inline contracts/Recruitment.sol:295]: Referrer will not be populated? see line 118...

File: contracts/Recruitment.sol [hunk 4]
```diff
@@ -427,18 +436,21 @@ contract Recruitment is Ownable, ReentrancyGuard {
...
 
-        bountyClaim[jobCandidatehire[_jobId].referrer] =
-            bountyClaim[jobCandidatehire[_jobId].referrer] +
-            (bounty * 6500) /
-            10_000;
-        bountyClaim[jobCandidatehire[_jobId].wallet] =
-            bountyClaim[jobCandidatehire[_jobId].wallet] +
-            (bounty * 1000) /
-            10_000;
+        uint hiredCount = hiredCpandidateListForJob[_jobId].length;
```
Comment (codingnuclei) [inline contracts/Recruitment.sol]: confirm with @Hangout2698 if we can hire multiple candidates for one job, and if yes - how we want to split bounty??

File: contracts/Recruitment.sol [hunk 5]
```diff
@@ -452,6 +464,63 @@ contract Recruitment is Ownable, ReentrancyGuard {
...
+        FrontDoorStructs.UserScore memory newScore = FrontDoorStructs.UserScore (
+          score,
+          msg.sender
+        );
+        candidateScores[candidateAddress].push(newScore);
+        referralScores[candidateList[candidateAddress].referrer].push(newScore);
+        updateCandidateScore(candidateAddress);
+        updateReferrerScore(candidateList[candidateAddress].referrer);
+    }
+    function setCompanyScoreFromCandidate(address companyAddress, uint256 score) public checkIfCandidateHiredByCompany(msg.sender , companyAddress) {
```
Comment (codingnuclei) [inline contracts/Recruitment.sol:484]: check formatting - each function should be separated by a new line.

File: hardhat.config.js [hunk 6]
```diff
@@ -21,10 +21,7 @@ module.exports = {
     hardhat: {
       initialBaseFeePerGas: 0
     },
-    sepolia: {
```
Comment (codingnuclei) [inline hardhat.config.js]: ?? why has this been removed

File: scoringTestStrings.txt
Comment (codingnuclei) [inline scoringTestStrings.txt:1]: test outputs should be added to `.gitIgnore`

**Review by codingnuclei — CHANGES_REQUESTED** [review]

File: contracts/Recruitment.sol (hunk 1 above)
Comment (sunnyalgo) [reply contracts/Recruitment.sol:53]: yeah, that's my type error.

Comment (sunnyalgo) [reply contracts/Recruitment.sol:53]: thx

File: contracts/Recruitment.sol (hunk 5 above)
Comment (sunnyalgo) [reply contracts/Recruitment.sol:478]: Yeah, I missed that. thx.

File: hardhat.config.js (hunk 6 above)
Comment (sunnyalgo) [reply hardhat.config.js:24]: my mistake, I tested on my local and not inserted again.

File: scoringTestStrings.txt
Comment (sunnyalgo) [reply scoringTestStrings.txt:1]: Nope, test results are made to show all of you.

File: contracts/Recruitment.sol (hunk 2 above)
Comment (sunnyalgo) [reply contracts/Recruitment.sol:115]: Sure. Thanks for your checking.

File: contracts/Recruitment.sol [hunk 7]
```diff
@@ -83,6 +84,11 @@ contract Recruitment is Ownable, ReentrancyGuard {
         require(isCompanyRegistered(_address), "Company is not registered yet");
         _;
     }
+    
+    modifier checkIfNewUser(address _address) {
```
Comment (codingnuclei) [inline contracts/Recruitment.sol:88]: one modifier per check i.e:

`checkIfRegisteredCandidate`
`checkIfRegisteredReferrer`
etc 

as i one will be able to apply for jobs and refer candidates.

**Review by codingnuclei — CHANGES_REQUESTED** [review]

File: contracts/Recruitment.sol (hunk 7 above)
Comment (sunnyalgo) [reply contracts/Recruitment.sol:88]: @Hangout2698 said me candidate can't apply for the job directly.

**Comment by fribas84** [comment]
Cannot Merge it, oudated versions

**Pull request closed without merging**